recipes/api_key.rb

```diff
@@ -0,0 +1,79 @@
+# Generate an API key
+api_key = nil
+ruby_block 'generate-api-key' do
+  block do
+    require 'net/https'
+    require 'http-cookie'
+    require 'json'
+    require 'securerandom'
+
+    hopsworks_fqdn = consul_helper.get_service_fqdn("hopsworks.glassfish")
+    _, hopsworks_port = consul_helper.get_service("glassfish", ["http", "hopsworks"])
+    if hopsworks_port.nil? || hopsworks_fqdn.nil?
+      raise "Could not get Hopsworks fqdn/port from local Consul agent. Verify Hopsworks is running with service name: glassfish and tags: [http, hopsworks]"
+    end
+
+    hopsworks_endpoint = "https://#{hopsworks_fqdn}:#{hopsworks_port}"
+    url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/auth/service")
+    api_key_url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/users/apiKey")
+
+    params =  {
+      :email => node['airflow']['user']['email'],
+      :password => node['airflow']['user']['pwd']
+    }
+
+    api_key_params = {
+      :name => "airflow_" + SecureRandom.hex(12),
+      :scope => "KAFKA&scope=FEATURESTORE&scope=PROJECT&scope=JOB&scope=DATASET_CREATE&scope=DATASET_VIEW&scope=SERVING&scope=DATASET_DELETE&scope=MODELREGISTRY&scope=GIT"
```

Comment:
No it should only have `JOB` scope. That's what Airflow should do, nothing else.